### PR TITLE
Feature add webhook

### PR DESCRIPTION
**Pull request by enishant**

### CODE REVIEW

File: Controller/BaseController.php [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 namespace Razorpay\Magento\Controller;
 
+// require in case of zip installation without composer
+require_once __DIR__ . "/../../Razorpay/Razorpay.php";
```
Comment (ramth05) [inline Controller/BaseController.php]: @enishant , this is not required we are adding this only in zip folder file as of now.

File: Controller/Payment/Webhook.php [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+                                . "PaymentId:(:$paymentId) failed with error: ". $e->getMessage());
+            return;
+        }
+        try {
+            # fetch the related sales order and verify the payment ID with rzp payment id
+            # To avoid duplicate order entry for same quote
+            $collection = $this->objectManagement->get('Magento\Sales\Model\Order')
+            ->getCollection()
+            ->addFieldToSelect('entity_id')
+            ->addFilter('increment_id', $quoteId)
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant , Variable name must be orderID. ( we are not using quote anymore here now).

File: Controller/Payment/Webhook.php [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+                        $order->addStatusHistoryComment(
+                            __(
+                                'Actual Amount %1 of %2, with Razorpay Offer/Fee applied.',
+                                "Authroized",
+                                $order->getBaseCurrency()->formatTxt($amountPaid)
+                            )
+                        );
+                        $order->save();
+                        //update quote
+                        $quote = $this->objectManagement->get('Magento\Quote\Model\Quote')
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant , this is not required as we are working with orders not the quotes.

File: Controller/Payment/Webhook.php [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+     * @param \Magento\Sales\Model\Service\InvoiceService $invoiceService
+     * @param \Magento\Framework\DB\Transaction $transaction
+     * @param \Magento\Catalog\Model\Session $catalogSession
+     * @param OrderRepositoryInterface $orderRepository
+     * @param \Magento\Sales\Model\Order\Email\Sender\InvoiceSender $invoiceSender
+     * @param \Magento\Sales\Model\Order\Email\Sender\OrderSender $orderSender
+     * @param \Magento\Framework\App\CacheInterface $cache
+     */
+    public function __construct(
+        \Magento\Framework\App\Action\Context $context,
```
Comment (ramth05) [inline Controller/Payment/Webhook.php:81]: @enishant , please verify if that much dependency needs to be loaded.

File: Controller/Payment/Webhook.php [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+     *
+     * @param array $post
+     */
+    protected function authorize(array $post)
+    {
+        $this->logger->info("Razorpay Webhook Event(" . $post['event'] . ") processing Started.");
+        $paymentId      = $post['payload']['payment']['entity']['id'];
+        $rzpOrderId     = $post['payload']['payment']['entity']['order_id'];
+        $amountPaid     = $post['payload']['payment']['entity']['amount'];
+        $email          = $post['payload']['payment']['entity']['email'];
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant , are we using it anywhere.

**Review by ramth05 — CHANGES_REQUESTED** [review]
@enishant , please test it with ngrok as well. Also take help from @jampulanaveen and test it together.

File: Controller/BaseController.php (hunk 1 above)
Comment (enishant) [reply Controller/BaseController.php]: @ramth05  Removed require_once as we are adding this only in release zip folder file.

File: Controller/Payment/Webhook.php (hunk 5 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05 Removed unused variables

File: Controller/Payment/Webhook.php (hunk 3 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05 Removed code written for quote update

File: Controller/Payment/Webhook.php (hunk 2 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05 Updated variable name to orderId

File: Controller/Payment/Webhook.php (hunk 4 above)
Comment (enishant) [reply Controller/Payment/Webhook.php:81]: @ramth05 Removed unused dependencies

**Pull request closed without merging**